ipykernel/ipkernel.py

```diff
@@ -274,21 +274,22 @@ def do_complete(self, code, cursor_pos):
 
     def _experimental_do_complete(self, code, cursor_pos):
         """
-        Experimental completions from IPython, using Jedi. 
+        Experimental completions from IPython, using Jedi.
         """
         if cursor_pos is None:
             cursor_pos = len(code)
         with provisionalcompleter():
             raw_completions = self.shell.Completer.completions(code, cursor_pos)
             completions = list(rectify_completions(code, raw_completions))
-            
+
             comps = []
             for comp in completions:
                 comps.append(dict(
                             start=comp.start,
                             end=comp.end,
                             text=comp.text,
                             type=comp.type,
+                            signature=getattr(comp, 'signature', '')
```

Comment:
Good Point! `signature` probably only makes sense in case of functions.
Atom's [autocomplete-plus](https://github.com/atom/autocomplete-plus/wiki/Provider-API#suggestions) uses the terms `rightLabel` and `leftLabel` so `right_label` would be another option.